query-node/mappings/src/workingGroup.ts

```diff
@@ -0,0 +1,180 @@
+
+  // call generic processing
+  await workingGroup_WorkerStorageUpdated(db, storageProviderId, newMetadata)
+}
+
+export async function storageWorkingGroup_TerminatedWorker(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId: storageProviderId} = new StorageWorkingGroup.TerminatedWorkerEvent(event).data
+
+  // call generic processing
+  await workingGroup_TerminatedWorker(db, storageProviderId)
+}
+
+export async function storageWorkingGroup_WorkerExited(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId: storageProviderId} = new StorageWorkingGroup.WorkerExitedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerExited(db, storageProviderId)
+}
+
+/////////////////// Gateway working group //////////////////////////////////////
+
+export async function gatewayWorkingGroup_OpeningFilled(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {applicationIdToWorkerIdMap} = new GatewayWorkingGroup.OpeningFilledEvent(event).data
+
+  // call generic processing
+  await workingGroup_OpeningFilled(db, StorageProviderType.GATEWAY, applicationIdToWorkerIdMap)
+}
+
+export async function gatewayWorkingGroup_WorkerStorageUpdated(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId: storageProviderId, bytes: newMetadata} = new GatewayWorkingGroup.WorkerStorageUpdatedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerStorageUpdated(db, storageProviderId, newMetadata)
+}
+
+export async function gatewayWorkingGroup_TerminatedWorker(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId: storageProviderId} = new StorageWorkingGroup.TerminatedWorkerEvent(event).data
+
+  // call generic processing
+  await workingGroup_TerminatedWorker(db, storageProviderId)
+}
+
+export async function gatewayWorkingGroup_WorkerExited(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId: storageProviderId} = new StorageWorkingGroup.WorkerExitedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerExited(db, storageProviderId)
+}
+
+/////////////////// Generic working group processing ///////////////////////////
+
+export async function workingGroup_OpeningFilled(
+  db: DatabaseManager,
+  storageProviderType: StorageProviderType,
+  applicationIdToWorkerIdMap: ApplicationIdToWorkerIdMap
+): Promise<void> {
+  const storageProviderIds = [...applicationIdToWorkerIdMap.values()]
+
+  for (const storageProviderId of storageProviderIds) {
+    // load storage provider
+    const storageProvider = await db.get(StorageProvider, { where: { id: storageProviderId.toString() } as FindConditions<StorageProvider> })
```

Comment:
This check is not necessary, the same worker id will never get re-used.
As a matter of fact if you find an existing worker id you have inconsistent state.

Reply:
The check has been removed.